android/src/main/java/com/thebluealliance/androidclient/helpers/PitLocationHelper.java

```diff
@@ -0,0 +1,201 @@
+package com.thebluealliance.androidclient.helpers;
+
+import com.thebluealliance.androidclient.R;
+import com.thebluealliance.androidclient.models.APIStatus;
+
+import android.content.Context;
+import android.preference.PreferenceManager;
+import android.util.JsonReader;
+
+import java.io.BufferedWriter;
+import java.io.File;
+import java.io.FileOutputStream;
+import java.io.FileReader;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.util.Date;
+import java.util.GregorianCalendar;
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * Encapsulation of special-cased stuff for showing pit locations at 2016 Champs
+ */
+public class PitLocationHelper {
+
+    private static final String LAST_UPDATED_PREF_KEY = "cmp_pit_locations_update_time";
+    private static final String PIT_LOCATIONS_FILENAME = "2016_pit_locations.json";
+
+    private static Map<String, TeamPitLocation> sTeamLocationCache = new HashMap<>();
+
+    public static boolean shouldShowPitLocation(Context context, String teamKey) {
+        Date champsStart = new GregorianCalendar(2016, GregorianCalendar.APRIL, 25).getTime();
+        Date champsEnd = new GregorianCalendar(2016, GregorianCalendar.MAY, 1).getTime();
+
+        Date now = new Date();
+        boolean isChamps = (now.after(champsStart) && now.before(champsEnd));
+        if (isChamps) {
+            return (getPitLocation(context, teamKey) != null);
+        }
+        return false;
+    }
+
+    public static TeamPitLocation getPitLocation(Context context, String teamKey) {
+        // First, check cache
+        if (sTeamLocationCache.containsKey(teamKey)) {
+            return sTeamLocationCache.get(teamKey);
+        }
+
+        // If that fails, try to load it into the cache
+        if (readTeamLocationFromJson(context, teamKey) != null) {
+            return sTeamLocationCache.get(teamKey);
+        }
+
+        // Didn't find anything
+        return null;
+    }
+
+    public static void updateFromRemoteUrl(Context context, String newContent, long updateTimeSeconds) {
+        // Write to the file
+        try {
+            File locationFile = getLocationsFile(context);
+            if (!locationFile.exists()) {
+                locationFile.createNewFile();
+            }
+
+            BufferedWriter writer = new BufferedWriter(new FileWriter(locationFile));
+            writer.write(newContent);
+            writer.flush();
+            writer.close();
+
+            // Note the update time, but only if it was successful
+            PreferenceManager.getDefaultSharedPreferences(context).edit().putLong(LAST_UPDATED_PREF_KEY, updateTimeSeconds).commit();
```

Comment:
I only used a Context because other parts of the code depend on the full Context object.